tiledb/sm/group/group_details_v1.cc

```diff
@@ -72,5 +70,44 @@ tdb_shared_ptr<Group> GroupV1::deserialize(
 
   return group;
 }
+
+Status GroupDetailsV1::apply_pending_changes() {
+  std::lock_guard<std::mutex> lck(mtx_);
+
+  // Remove members first
+  for (const auto& member : members_to_modify_) {
+    auto& uri = member->uri();
+    if (member->deleted()) {
+      members_.erase(uri.to_string());
```

Comment:
What happens if the user calls delete more than once?

Reply:
When erasing by key, if no key exists it return 0 instead of 1. Its a safe operation so this can be allowed:
https://cplusplus.com/reference/unordered_map/unordered_map/erase/